plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/refactoring/wizards/rename/PyRenameGlobalProcess.java

```diff
@@ -39,7 +46,32 @@ protected boolean getRecheckWhereDefinitionWasFound() {
             String initialName, SourceModule module) {
         SimpleNode searchStringsAt = module.getAst();
 
-        List<ASTEntry> ret = ScopeAnalysis.getLocalOccurrences(initialName, module.getAst());
+        List<String> modNames = StringUtils.split(request.moduleName, '.');
+        List<String> actualModNames = StringUtils.split(module.getName(), '.');
+        FastStringBuffer buf = new FastStringBuffer();
+        int i = 0;
+        while (i < modNames.size() && i < actualModNames.size() && modNames.get(i).equals(actualModNames.get(i))) {
+            i++;
+        }
+        while (i < modNames.size()) {
+            buf.append(modNames.get(i)).append('.');
+            i++;
+        }
+        buf.deleteLast();
+        String compare = buf.toString();
+        boolean valid = false;
+        for (stmtType content : NodeUtils.getBody(searchStringsAt)) {
```

Comment:
This doesn't seem ideal... there are a few reasons:
1. You're doing a pre-check on the places to rename, but in case it matches it shouldn't switch the whether to find all parts (it could go on and create the entry directly).
2. You're just visiting top-level imports (so, if there's an import inside a method it won't be gotten by this.

I think that the logic to check the import should be done inside of `getLocalOccurrences` itself when `onlyFirstAttribPart==false` (it's visitor structure should already visit children nodes).